homeassistant/components/modbus/light.py

```diff
@@ -32,9 +52,172 @@ async def async_setup_platform(
 class ModbusLight(BaseSwitch, LightEntity):
     """Class representing a Modbus light."""
 
-    _attr_color_mode = ColorMode.ONOFF
-    _attr_supported_color_modes = {ColorMode.ONOFF}
+    def __init__(
+        self, hass: HomeAssistant, hub: ModbusHub, config: dict[str, Any]
+    ) -> None:
+        """Initialize the Modbus light entity."""
+        super().__init__(hass, hub, config)
+        self._attr_brightness: int = LIGHT_DEFAULT_BRIGHTNESS
+        self._brightness_address: int | None = config.get(CONF_BRIGHTNESS_REGISTER)
+        self._color_temp_address: int | None = config.get(CONF_COLOR_TEMP_REGISTER)
+        self._attr_color_temp_kelvin: int | None = None
+
+        # Determine color mode dynamically
+        self._attr_color_mode = self._detect_color_mode(config=config)
```

Comment:
```suggestion
        self._attr_color_mode = self._detect_color_mode(config)
```
no advantage in using kwargs here